lib/spack/spack/build_environment.py

```diff
@@ -469,7 +468,7 @@ def set_wrapper_variables(pkg, env):
         env.set(SPACK_DEBUG, "TRUE")
     env.set(SPACK_SHORT_SPEC, pkg.spec.short_spec)
     env.set(SPACK_DEBUG_LOG_ID, pkg.spec.format("{name}-{hash:7}"))
-    env.set(SPACK_DEBUG_LOG_DIR, spack.paths.spack_working_dir)
+    env.set(SPACK_DEBUG_LOG_DIR, pkg.stage.path)
```

Comment:
This is the only reference to `pkg.stage` in `build_environment`, but I think stages always exist when `set_wrapper_variables` is run:  i.e. during building/testing:

```
    need_compiler = context == Context.BUILD or (
        context == Context.TEST and pkg.test_requires_compiler
    )
    if need_compiler:
```

(moreover, referring to `pkg.stage` does not appear to create the stage, although I don't know if any of our tests enforce that)